src/java/org/apache/cassandra/locator/AbstractCloudMetadataServiceConnector.java

```diff
@@ -61,6 +80,7 @@ public String apiCall(String url,
             conn = (HttpURLConnection) new URL(url + query).openConnection();
             extraHeaders.forEach(conn::setRequestProperty);
             conn.setRequestMethod(method);
+            conn.setConnectTimeout(30000);
```

Comment:
Should we take the timeout from the properties with 30s as default?

Reply:
well it is tricky to pass anything to these connectors from the properties as currently it is accepting metadata service url only. We might either refactor bunch of stuff for such a triviality or we can just have a system property in `CassandraRelevantProperties`? It is hard to believe this would be something which would be tweaked by users regularly, maybe for a power users as the ultimate way to weak it but otherwise ....

Reply:
since we have a generic metadata_url - why not just have a generic metadata_request_timeout in snitch properties? Though, I don't insist on this one 😃

Reply:
nobody is ever going to configure it, we dont need this

Reply:
Anyway, I think I will do it in such a way that we will pass SnitchProperties to connector. Because my gut feeling tells me that there might be connectors which might be somehow further configured. Now it is pretty simple, just metadata url and possibly timeout, but what if connectors need something fancy? I dont know ... certificates? You feel me. So the needs to be a way how to get the access to all snitch properties.